Added config variable to control if updateJSON needs to fail when missing variable

robertvanlienden opened the pull request:
After migrating `sheadawson/silverstripe-linkable` to `gorriecoe/silverstripe-linkfield` I've was unable to add an Link to the SiteConfig.
Got the following exception;
`VARIABLE doesn't exist in POST data.`.
This exception came out of the `JSONTextExtension` of this module.

Seems that when you use `gorriecoe/silverstripe-linkfield` in combination with `JSONTextExtension` on `SiteConfig`, when you save the link, the `onBeforeWrite` => `updateJSON` gets triggered.

Because the Linkfield-form you are submitting (in the context of SiteConfig) not containing the JSON data, the exception comes up.

Fixed this by adding an config variable where you can disable the exception to be thrown;
```
Silverstripe\SiteConfig\SiteConfig:
  jsonTextExtensionFailOnMissingKey: false
```

This change seems to make everything work again, so decided to open an PR
Because of the way I've implemented this, this can be added without breaking changes :smile: .

robertvanlienden committed:
Added config variable to control if updateJSON needs to fail when variable not set

erikfrerejean reviewed (changes requested):
The `|| false` seems to be a left over debug statement.

robertvanlienden commented:
> The `|| false` seems to be a left over debug statement.

@erikfrerejean This was not intended to be an debug statement, but something for backwards compatibility.
If no config variable is set, I want the old behavior. 

* edited *

My bad. Seems like I was confused between the behavior I wanted (that I could add an link) and the behavior I wanted for this module.
Fixed it in the lasted commit.

Replaced the if statement with the following;
```
                    if ($this->owner->config()->get('jsonTextExtensionFailOnMissingKey') !== null) {
                        $failOnMissingKey = $this->owner->config()->get('jsonTextExtensionFailOnMissingKey');
                    } else {
                        $failOnMissingKey = false;
                    }

                    if (!$failOnMissingKey) {
```

This will add some more code, but is readable and will for sure work in PHP5.6 :wink:

robertvanlienden committed:
Replaced '|| false' with some if statement. More readable, and always working.

phptek commented:
Just a note to say "thanks"! And that I have spotted this activity and will have a nosey soon. Apologies, I was out of the country with sketchy internet access.

dcentrica commented:
Released as `1.0.3`. Thanks for your contribution and apologies it took so long to merge.
